src/video/surface.c

```diff
@@ -1,48 +1,54 @@
 #include "video/surface.h"
 #include "utils/allocator.h"
 #include "utils/miscmath.h"
+#include <video.h>
+#if !defined(N64_BUILD) && !defined(MIN_BUILD)
 #include "utils/png_writer.h"
+#endif
 #include <stdlib.h>
+#include <SDL.h>
 
 // Each surface is tagged with a unique key. This is then used for texture atlas.
 // This keeps track of the last index used.
 static unsigned int guid = 0;
 
-void surface_create(surface *sur, int w, int h) {
+void surface_create(surface *sur, int w, int h, int transparent) {
+#ifdef N64_BUILD
+    sur->data = malloc_uncached_aligned(64, w * h);
```

Comment:
Instead of making different reservation for each platform, we should probably have our own omf_malloc_uncached_aligned() or somesuch. Then x86 could just map to alloc, while N64 could map to malloc_uncached_aligned. That would also satisfy the requirement to have memory allocation wrapped in our omf_prefixed namespace.

Reply:
It's slight complicated and not sure if we need to deal with it. The N64 is a UMA (unified memory architecture) to some extent, so memory allocated can be used as a texture, but ofcourse there are some requirements on alignment because the hardware has less than 32 address bits.

So here it makes sense for surfaces, other platforms will be able to use malloc normally unless they want to allocate.
The problem is also that libdragon requires a matching aligned alloc to have a matching aligned free. Becasue it stores some header info somewhere.

Again very specific, I would wait until we have another platform that needs something different.

I also don't see x86 allocating GPU memory here because then all accesses would need to have a map/unmap call.

Reply:
x86 would not allocate gpu memory. X86 would just use normal malloc(). The point is, we would have the omf_malloc_uncached_aligned(), which would only do something specific on platforms that need it to do something specific. Then we could use that omg_malloc_uncached_aligned() only on sprites and stuff that need it.

The issue with waiting until someone else needs it, is that then the response to this same query will be "But it's already done like this, and I don't know about n64, and it's too much work" :(

Reply:
Right, you tag the allocation as potentially needing special handling, and each platform can decide if it needs to do anything special.

Reply:
I think what would make sense is surface_create, surface_free, sd_vga_image_create,sd_vga_image_free move to platform.c.

platform.c will have a different implementation based on the platform. 
I will clean that up when the N64 files start coming in.

Reply:
No, those functions should really stay where they are, as that is where the devs will look for them and that is what makes sense. There is no point in moving them anyway, because it's not the right abstraction level. Abstracting malloc is.

Reply:
Then this function will still need an ifdef N64_BUILD. Because the alignment is platform specific.